src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -453,42 +485,30 @@ private boolean addProvisionedSlave(@Nonnull PodTemplate template, @CheckForNull
         }
 
         PodList slaveList = client.pods().inNamespace(templateNamespace).withLabels(getLabels()).list();
-        List<Pod> slaveListItems = slaveList.getItems();
+        List<Pod> allActiveSlavePods = slaveList.getItems().stream()
+                .filter(x -> x.getStatus().getPhase().toLowerCase().matches("(running|pending)"))
+                .collect(Collectors.toList());
 
         Map<String, String> labelsMap = new HashMap<>(this.getLabels());
         labelsMap.putAll(template.getLabelsMap());
-        PodList namedList = client.pods().inNamespace(templateNamespace).withLabels(labelsMap).list();
-        List<Pod> namedListItems = namedList.getItems();
-
-        if (this.isCapOnlyOnAlivePods()) {
-            slaveListItems = slaveListItems.stream()
-                                           .filter(x -> x.getStatus()
-                                                         .getPhase().toLowerCase()
-                                                                    .matches("(running|pending)"))
-                                           .collect(Collectors.toList());
-        }
-
-        if (template.isCapOnlyOnAlivePods()) {
-            namedListItems = namedListItems.stream()
-                                           .filter(x -> x.getStatus()
-                                                         .getPhase().toLowerCase()
-                                                                    .matches("(running|pending)"))
-                                           .collect(Collectors.toList());
-        }
+        PodList templateSlaveList = client.pods().inNamespace(templateNamespace).withLabels(labelsMap).list();
+        List<Pod> allTemplateSlavePods = templateSlaveList.getItems().stream()
+                .filter(x -> x.getStatus().getPhase().toLowerCase().matches("(running|pending)"))
```

Comment:
the name `allTemplateSlavePod` implies to me that you would *NOT* filter based on phase .... can you reconcile either with removing the filter, changing the var name, comment explaining what is up, etc.